contracts/MultisendNative.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.18;
+
+/**
+ * @dev Taraxa Multisend Contract
+ *
+ * The Multisend Contract is used to do multiple native token transfers in the same transaction
+ */
+contract MultisendNative {
+    /**
+     * @dev fallback receiver
+     */
+    receive() external payable {}
+
+    /**
+     * @dev Transfers the tokens from the method caller to the participant.
+     * @return (lastSentIndex, totalSent) -> the index of the last sent and the total value.
+     * The method doesn't restrict the sender to strictly be abel to fill all the transactions
+     * as it will simply end the loop once there isn't enough balance for the next transfer.
+     *
+     * Emits a {TokensSent} event.
+     */
+    function multisendToken(address payable[] calldata _recipients, uint256[] calldata _amounts)
+        public
+        payable
+        returns (uint8, uint256)
+    {
+        require(_recipients.length <= 200, 'Multisend: max transfers per tx exceeded');
+        require(
+            _recipients.length == _amounts.length,
+            'Multisend: contributors and balances have different sizes'
+        );
+        require(msg.value >= _amounts[0], "Multisend: Sent value doesn't cover first transaction");
+
+        uint256 total = 0;
+        uint8 i = 0;
+        for (i; i < _recipients.length; ++i) {
+            if ((msg.value - total) >= _amounts[i]) {
+                _recipients[i].transfer(_amounts[i]);
+                total = total + _amounts[i];
+            } else {
+                if (msg.value > total) {
+                    payable(msg.sender).transfer(msg.value - total);
+                    emit SentBack(msg.value - total);
+                }
+                break;
```

Comment:
Too complicated and doesn't cover the case where we send this function more tokens than in the amounts array.

Let's just break here and move the send back functionality after the for